services/route_manager/src/main/java/com/futurewei/alcor/route/service/Impl/RouteWithVpcMapperServiceImpl.java

```diff
@@ -0,0 +1,154 @@
+import com.futurewei.alcor.route.dao.RouteWithVpcMapperRepository;
+import com.futurewei.alcor.route.service.RouteWithVpcMapperService;
+import com.futurewei.alcor.web.entity.route.RouteEntity;
+import com.futurewei.alcor.web.entity.route.VpcToRouteMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Service
+public class RouteWithVpcMapperServiceImpl implements RouteWithVpcMapperService {
+
+    private Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    @Autowired
+    RouteWithVpcMapperRepository routeWithVpcMapperRepository;
+
+    @Autowired
+    RouteRepository routeRepository;
+
+    @Override
+    public VpcToRouteMapper getByVpcId(String vpcId) throws ResourceNotFoundException, ResourcePersistenceException {
+        try {
+            return this.routeWithVpcMapperRepository.findItem(vpcId);
+        }catch (Exception e) {
+            return null;
+        }
+    }
+
+    @Override
+    public List<RouteEntity> getRuleByVpcId(String vpcId) throws ResourceNotFoundException, ResourcePersistenceException {
+
+        List<RouteEntity> routes = new ArrayList<>();
+
+        try {
+            VpcToRouteMapper vpcToRouteMapper =  this.routeWithVpcMapperRepository.findItem(vpcId);
+            if (vpcToRouteMapper == null) {
+                return null;
+            }
+
+            List<String> routeIds = vpcToRouteMapper.getRouteIds();
+            for (String routeId : routeIds) {
+                RouteEntity route = this.routeRepository.findItem(routeId);
+                if (route != null) {
+                    routes.add(route);
+                }
+            }
+
+            return routes;
+
+        }catch (Exception e) {
+            return null;
+        }
+    }
+
+    @Override
+    public Map getAllMappers() throws CacheException {
+        return this.routeWithVpcMapperRepository.findAllItems();
+    }
+
+    @Override
+    public void addMapper(VpcToRouteMapper vpcToRouteMapper) throws DatabasePersistenceException {
+        try {
+            this.routeWithVpcMapperRepository.addItem(vpcToRouteMapper);
+        } catch (Exception e) {
+            throw new DatabasePersistenceException(e.getMessage());
+        }
+    }
+
+    @Override
+    public void addMapperByRouteEntity(String vpcId, RouteEntity routeEntity) throws DatabasePersistenceException {
+        try {
+            if (routeEntity == null) {
+                return;
+            }
+
+            String routeId = routeEntity.getId();
+            VpcToRouteMapper vpcToRouteMapper = getByVpcId(vpcId);
+            if (vpcToRouteMapper == null) {
+                vpcToRouteMapper = new VpcToRouteMapper();
+                vpcToRouteMapper.setVpcId(vpcId);
+            }
+            List<String> routeIds = vpcToRouteMapper.getRouteIds();
```

Comment:
We need to run Ctrl+Shift+L for proper code format. Seems that at least this file doesn't follow the formatting guidance.

Reply:
Sure

Reply:
Sure